pyintegration/deephaven2/dtypes.py

```diff
@@ -99,31 +108,236 @@ def array_from(self, values: Iterable):
+            if el is None:
+                return NULL_CHAR
+            if isinstance(el, int):
+                return el
+            if isinstance(el, str):
+                if len(el) < 1:
+                    return NULL_CHAR
+                return ord(el[0])
+            try:
+                return int(el)
+            except ValueError:
+                return NULL_CHAR
+
+        try:
+            if isinstance(seq, str):
+                return super().array_of([ord(c) for c in seq])
+            elif isinstance(seq, np.ndarray):
+                if seq.dtype == np.uint16:
+                    return super().array_of(seq)
+                elif np.issubdtype(seq.dtype, np.integer):
+                    return super().array_of(seq.astype(np.uint16))
+                elif seq.dtype == np.dtype('U1') and seq.dtype.name in ['unicode32', 'str32', 'string32',
+                                                                        'bytes32']:
+                    arr = np.copy(seq)
+                    arr.dtype = np.uint32
+                    return super().array_of(arr.astype(np.uint16))
+                elif seq.dtype == np.dtype('S1') and seq.dtype.name in ['str8', 'string8', 'bytes8']:
+                    arr = np.copy(seq)
+                    arr.dtype = np.uint8
+                    return super().array_of(arr.astype(np.uint16))
+                elif seq.dtype == object:
+                    return super().array_of(np.array([to_char(el) for el in seq], dtype=np.uint16))
+                else:
+                    # do our best
+                    raise ValueError(
+                        f"Passed in a numpy array, expect integer dtype or one char string dtype, and got {seq.dtype}")
+            elif isinstance(seq, pd.Series):
+                return self.array_of(seq.values)
+            else:
+                return self.array_of(np.asarray(seq))
+        except Exception as e:
+            raise DHError(e, f"failed to create a Java {self.j_name} array.") from e
+
+
+# region convenience enum and module level type aliases
+class DTypes(Enum):
+    """ An enum that defines a predefined set of supported data types in Deephaven. """
+
+    bool_ = DType(j_name="java.lang.Boolean", qst_type=_JQstType.booleanType(), )
+    byte = IntegerDType(j_name="byte", qst_type=_JQstType.byteType(), np_dtypes=(np.int8, np.uint8),
+                        null_value=NULL_BYTE)
+    int8 = byte
+    short = IntegerDType(j_name="short", qst_type=_JQstType.shortType(), np_dtypes=(np.int16, np.uint16),
+                         null_value=NULL_SHORT)
+    int16 = short
+    char = CharDType(j_name="char", qst_type=_JQstType.charType(), np_dtypes=(), null_value=NULL_CHAR)
+    int_ = IntegerDType(j_name="int", qst_type=_JQstType.intType(), np_dtypes=(np.int32, np.uint32),
+                        null_value=NULL_INT)
+    int32 = int_
+    long = IntegerDType(j_name="long", qst_type=_JQstType.longType(), np_dtypes=(np.int64, np.uint64),
```

Comment:
`np.unit64` is an interesting case, since we can't promote it to a larger type to avoid truncation.  thoughts?